### PR TITLE
feat: Judge Middleware

### PR DESCRIPTION
**Pull request by Ansaar-Syed**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

File: packages/api/src/middleware/judgeMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from 'express';
+import { Judge } from '@hangar/database';
+
+/**
+ * Middleware that evaluates the judge's validity
+ *
+ * Paths:
+ *  - Next function invoked: judge wa identified and matching onject mounted to request
```
Comment (SpencerKaiser) [inline packages/api/src/middleware/judgeMiddleware.ts:8]:
```suggestion
 *  - Next function invoked: judge was identified and matching object mounted to request
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Perfect! Just need tests!

File: packages/api/src/middleware/judgeMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (judge) {
+    req.judge = judge;
+  } else {
+    // Judge does not exist in database
+    res.sendStatus(403);
+    return;
+  }
+
+  next();
```
Comment (SpencerKaiser) [inline packages/api/src/middleware/judgeMiddleware.ts:30]: Just a slightly cleaner look
```suggestion
  if (!judge) {
    // Judge does not exist in database
    res.sendStatus(403);
    return;
  }
  
  req.judge = judge;
  next();
```

Comment (Ansaar-Syed) [reply packages/api/src/middleware/judgeMiddleware.ts:30]: got it , cam i commit it from here itself? @SpencerKaiser

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

File: packages/api/tests/middleware/judgeMiddleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const mockNext = jest.fn();
+
+    await judgeMiddleware(mockReq as any, mockRes as any, mockNext);
+    expect(entityManager.findOne).toBeCalledTimes(1);
+    expect(entityManager.findOne).toBeCalledWith(
+      expect.anything(),
+      expect.objectContaining({ user: mockSession.id }),
+    );
+    expect(mockRes.sendStatus).not.toHaveBeenCalled();
+    expect(mockNext).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (Ansaar-Syed) [reply packages/api/tests/middleware/judgeMiddleware.test.ts:22]: hmm, how would that be?

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**